util/test/demos/d3d12/d3d12_interleaved_multi_queue.cpp

```diff
@@ -0,0 +1,459 @@
+      "queues that may initially appear blocked, waiting for a fence signal "
+      "from another queue.";
+
+  enum CBVUAVSRVDescriptorHeapIdx : uint32_t
+  {
+    eBufferUAV = 0,
+    eBufferSRV,
+  };
+
+  std::string sources = R"EOSHADERS(
+
+cbuffer RootConstants : register(b0)
+{
+  uint rootConstant0;
+  uint rootConstant1;
+};
+RWStructuredBuffer<float4> bufferRW : register(u0);
+StructuredBuffer<float4> buffer : register(t0);
+
+[numthreads(1,1,1)]
+void mainCS()
+{
+  uint myBufferIdx = rootConstant0;
+  float4 myCol = float4((float)(rootConstant1 & 0xffu) / 255.0f, (float)((rootConstant1>>8u) & 0xffu) / 255.0f, (float)((rootConstant1>>16u) & 0xffu) / 255.0f, (float)((rootConstant1>>24u) & 0xffu) / 255.0f);
+  bufferRW[myBufferIdx] = myCol;
+}
+
+float4 mainVS(in float3 pos : POSITION) : SV_POSITION
+{
+	return float4(pos, 1);
+}
+
+float4 mainPS(in float4 pos : SV_POSITION) : SV_Target0
+{
+  uint myBufferIdx = rootConstant0;
+  return buffer[myBufferIdx];
+}
+
+)EOSHADERS";
+
+  enum class CBVUAVSRVHeapIdx : uint32_t
+  {
+    eBufferUAV = 0,
+    eBufferSRV
+  };
+
+  int main()
+  {
+    // initialise, create window, create device, etc
+    if(!Init())
+      return 3;
+
+    ComputeQueues<2> computeQs(dev);
+
+    ID3DBlobPtr csblob = Compile(sources, "mainCS", "cs_5_0");
+    ID3DBlobPtr vsblob = Compile(sources, "mainVS", "vs_4_0");
+    ID3DBlobPtr psblob = Compile(sources, "mainPS", "ps_4_0");
+
+    const Vec3f twoTrisVBDat[6] = {
+        Vec3f(-0.5f, -0.25f, 0.9f), Vec3f(0.25f, 0.5f, 0.9f),   Vec3f(1.0f, -0.25f, 0.9f),
+        Vec3f(-0.75f, 0.75f, 0.5f), Vec3f(0.75f, -0.75f, 0.5f), Vec3f(-0.75f, -0.75f, 0.5f),
+    };
+    ID3D12ResourcePtr vb = MakeBuffer().Data(twoTrisVBDat);
+
+    const D3D12_DESCRIPTOR_RANGE1 tableDescriptorRanges[2] = {
+        {
+            D3D12_DESCRIPTOR_RANGE_TYPE_UAV,
+            1,    // NumDescriptors
+            0,    // BaseShaderRegister
+            0,    // RegisterSpace
+            D3D12_DESCRIPTOR_RANGE_FLAG_DATA_VOLATILE,
+            (uint32_t)CBVUAVSRVHeapIdx::eBufferUAV    // OffsetInDescriptorsFromTableStart
+        },
+        {
+            D3D12_DESCRIPTOR_RANGE_TYPE_SRV,
+            1,                                            // NumDescriptors
+            0,                                            // BaseShaderRegister
+            0,                                            // RegisterSpace
+            D3D12_DESCRIPTOR_RANGE_FLAG_DATA_VOLATILE,    // D3D12_DESCRIPTOR_RANGE_FLAG_NONE
+            (uint32_t)CBVUAVSRVHeapIdx::eBufferSRV        // OffsetInDescriptorsFromTableStart
+        }};
+    const D3D12_ROOT_PARAMETER1 descriptorTableRootParam = {
+        D3D12_ROOT_PARAMETER_TYPE_DESCRIPTOR_TABLE,
+        {// DescriptorTable / D3D12_ROOT_DESCRIPTOR_TABLE1
+         sizeof(tableDescriptorRanges) / sizeof(tableDescriptorRanges[0]),    // NumDescriptorRanges
+         &tableDescriptorRanges[0]},
+        D3D12_SHADER_VISIBILITY_ALL};
```

Comment:
I've made the change from a single root descriptor table of 2-ranges to 2 root tables (using `tableParam`).

I can understand if you prefer to use multiple instances of `tableParam`, especially when root parameters aren't at a premium, which is the case here.  I did it that way initially just because that seemed the most usual way of doing it and with the slightly unusual way `tableParam` builds up hidden internal table ranges, I couldn't actually get the exact descriptor table I had in mind ... but I'm fine with changing the structure a little to cut down on a few lines.